agi-bin/directory.lib.php

```diff
@@ -187,6 +187,21 @@ function readContact($con, $keys='#'){
 					}
 				}
 				//fallthough if not successfull
+			case 'tts':
+				// speak the name if possible, otherwise move on to spell it
+				$tmp = $this->agi_get_var('ASTSPOOLDIR') . '/tmp/directory-tts-' . time() . rand(100, 999);
+				system('flite -t "' . $con['name'] .'" -o ' . $tmp . '.wav', $exit_code);
```

Comment:
$con['name'] needs to be run through: http://php.net/manual/en/function.escapeshellarg.php